Sources/Apodini/Semantic Model Builder/InterfaceExporter/InterfaceExporter.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Created by Andi on 22.11.20.
+//
+
+import class Vapor.Application
+import protocol NIO.EventLoop
+
+/// The Protocol any Exporter Request type must conform to
+protocol ExporterRequest {}
+
+/// When your `ExporterRequest` conforms to this protocol, it indicates that it delivers
+/// its own `EventLoop` out of the box. Having that conformance you can use a shorthand
+/// `EndpointRequestHandler.handleRequest(...)` method on without specifying an `EventLoop`.
+protocol WithEventLoop {
+    var eventLoop: EventLoop { get }
+}
+
+func null<T>(_ type: T.Type = T.self) -> T? {
+    T?(nil)
+}
+
+/// Any Apodini Interface Exporter must conform to this protocol.
+protocol InterfaceExporter {
+    /// Defines the type of the Request the exporter uses.
+    associatedtype ExporterRequest: Apodini.ExporterRequest
+    /// Defines the return type of the `export` method. The return type is currently unused.
+    associatedtype EndpointExportOutput = Void
+    /// Defines the return ype of the `exportParameter` method. For more details see `exportParameter(...)`
+    associatedtype ParameterExportOutput = Void
+
+    init(_ app: Application)
+
+    /// This method is called for every `Endpoint` on start up, which must be exporter
+    /// by the `InterfaceExporter`.
+    ///
+    /// - Parameter endpoint: The `Endpoint` which is to be exported.
+    /// - Returns: `EndpointExportOutput` which is defined by the `InterfaceExporter`.
+    func export<C: Component>(_ endpoint: Endpoint<C>) -> EndpointExportOutput
+
+    /// This optional method can be defined to export a `EndpointParameter`.
+    /// It is called for every `EndpointParameter` on an `Endpoint` when calling `Endpoint.exportParameters(...)`.
+    /// `Endpoint.exportParameters(...)` returns an array of `ParameterExportOutput`, being what is returned by this method.
+    ///
+    /// - Parameter parameter:
+    /// - Returns: `ParameterExportOutput` which is defined by the `InterfaceExporter`
+    func exportParameter<Type: Codable>(_ parameter: EndpointParameter<Type>) -> ParameterExportOutput
```

Comment:
Do you have a larger use-case for this method in mind?
The current example of the `RESTInterfaceExporter` returning the name and then only using it for logging seems a bit artificial to me, and I am not sure what to use this for instead.
The added type information for the parameter is great, but we should only add it if we really need it.

Reply:
An exporter might need a complete model of the endpoint‘s input to process incoming messages. This would be the place to create this model I guess. My current version of the websocket exporter would use this function (at least until the shared model takes over input validation).

Reply:
> Do you have a larger use-case for this method in mind?

As Max said, I think this a great way of building your internal Parameter models. Its the only way of accessing the generic type of the EndpointParameter. As somebody might need to access that type information in a generic way, when exporting its interface,  I came up with that as a solution.
But its optional to implement, meaning there is a empty default implementation. So if you don't need it, don't use it.

That was the thought behind it.

> The current example of the RESTInterfaceExporter returning the name and then only using it for logging seems a bit artificial to me, and I am not sure what to use this for instead.

It is. 😅 This was only there to demonstrate how it is called and how you can access the returned values.

Reply:
This makes sense. Thanks for the clarification.
Based on this, I think we can utilize the `exportParameter` for the gRPC exporter. Just didn't occur to me while looking through the PR 😁

Reply:
👍